app/modules/import-export/assets/js/admin.js

```diff
@@ -1,19 +1,231 @@
+	 *
+	 * @type {string}
+	 */
+	KIT_DATA_KEY = 'elementor-kit-data';
 
-if ( revertButton ) {
-	revertButton.addEventListener( 'click', ( event ) => {
+	/**
+	 * Contains the ID of the referrer Kit and the name of the Kit to remove. Stored in session storage.
+	 *
+	 * @type {Object}
+	 */
+	cachedKitData;
+
+	/**
+	 * The 'Remove Kit' revert button
+	 *
+	 * @type {Element}
+	 */
+	revertBtn;
+
+	/**
+	 * Name of the kit currently active (last imported)
+	 *
+	 * @type {string}
+	 */
+	activeKitName;
+
+	constructor() {
+		this.activeKitName = this.getActiveKitName();
+		this.revertBtn = document.getElementById( 'elementor-import-export__revert_kit' );
+
+		if ( this.revertBtn ) {
+			this.revertBtn.addEventListener( 'click', this.revertBtnOnClick.bind( this ) );
+			this.maybeAddRevertBtnMargin();
+		}
+
+		this.maybeShowReferrerKitDialog();
+	}
+
+	/**
+	 * Add bottom margin to revert btn if referred from Kit library
+	 */
+	maybeAddRevertBtnMargin() {
+		const referrerKitId = new URLSearchParams( this.revertBtn.href ).get( 'referrer_kit' );
+		if ( ! referrerKitId ) {
+			return;
+		}
+
+		this.revertBtn.style.marginBottom = this.calculateMargin();
+		this.scrollToBottom();
+	}
+
+	/**
+	 * CalculateMargin
+	 *
+	 * @return {string}
+	 */
+	calculateMargin() {
+		const adminBar = document.getElementById( 'wpadminbar' );
+		const adminBarHeight = adminBar ? adminBar.offsetHeight : 0;
+
+		const revertKitHeight = this.revertBtn.parentElement.offsetHeight;
+
+		return (
+			document.body.clientHeight -
+			adminBarHeight -
+			revertKitHeight -
+			document.getElementById( 'wpfooter' ).offsetHeight -
+			15 // Extra margin at the top
+		) + 'px';
+	}
+
+	/**
+	 * Scroll to the bottom of the page
+	 */
+	scrollToBottom() {
+		setTimeout( () => {
+			window.scrollTo( 0, document.body.scrollHeight );
+		} );
+	}
+
+	/**
+	 * RevertBtnOnClick
+	 *
+	 * @param {Event} event
+	 */
+	revertBtnOnClick( event ) {
 		event.preventDefault();
 
 		elementorCommon.dialogsManager.createWidget( 'confirm', {
-			headerMessage: __( 'Sure you want to make these changes?', 'elementor' ),
-			message: __( 'Removing assets or changing your site settings can drastically change the look of your website.', 'elementor' ),
+			headerMessage: __( 'Are you sure?', 'elementor' ),
+			message: __( 'Removing ', 'elementor' ) + this.activeKitName + __( ' will permanently delete changes made to the Kit\'s content and site settings', 'elementor' ),
+			strings: {
+				confirm: __( 'Delete', 'elementor' ),
+				cancel: __( 'Cancel', 'elementor' ),
+			},
+			onConfirm: this.onRevertConfirm,
+		} ).show();
+	}
+
+	onRevertConfirm() {
+		const referrerKitId = new URLSearchParams( this.revertBtn.href ).get( 'referrer_kit' );
+
+		this.saveToCache( referrerKitId ?? '' );
+
+		location.href = this.revertBtn.href;
+	}
+
+	maybeShowReferrerKitDialog() {
+		const { referrerKitId } = this.getDataFromCache();
+		if ( undefined === referrerKitId ) {
+			return;
+		}
```

Comment:
do you really need the undefined?

Reply:
yes this is one of three cases, the case where we don't show a dialog.
We can't check for a falsey value because that the second case is for an empty string (false value) which shows the non-referred dialog